iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBStartCheck.java

```diff
@@ -333,4 +347,23 @@ public void generateOrOverwriteSystemPropertiesFile() throws IOException {
     systemProperties.forEach((k, v) -> properties.setProperty(k, v.get()));
     systemPropertiesHandler.overwrite(properties);
   }
+
+  public void checkEncryptMagicString() throws IOException, ConfigurationException {
+    properties = systemPropertiesHandler.read();
+    String encryptMagicString = properties.getProperty("encrypt_magic_string");
+    if (encryptMagicString != null) {
+      byte[] magicBytes = EncryptUtils.hexStringToByteArray(encryptMagicString);
+      if (Objects.equal(magicString, new String(magicBytes))) {
+        serializeEncryptMagicString();
+        return;
+      }
+      String newMagicString =
+          new String(EncryptUtils.getEncrypt().getDecryptor().decrypt(magicBytes));
```

Comment:
Will changing the key cause decrypt() to throw an exception?
If it will, it's better to catch the exception and convert it to an understandable one.